src/main/java/com/gmail/yuramitryahin/security/UserDetailsServiceImp.java

```diff
@@ -0,0 +1,44 @@
+package com.gmail.yuramitryahin.security;
+
+import com.gmail.yuramitryahin.model.Role;
+import com.gmail.yuramitryahin.model.User;
+import com.gmail.yuramitryahin.service.UserService;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDetailsServiceImp implements UserDetailsService {
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserDetailsServiceImp(UserService userService, PasswordEncoder passwordEncoder) {
+        this.userService = userService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() ->
+                new UsernameNotFoundException("User not found!"));
+        org.springframework.security.core.userdetails.User.UserBuilder builder = null;
+        builder = org.springframework.security.core.userdetails.User.withUsername(username);
```

Comment:
We can use static import for this method

Reply:
fixed